Region/Histogram.cc

```diff
@@ -30,23 +31,23 @@ void Histogram::join(Histogram& h) { // NOLINT
     auto loop = [this, &h](const tbb::blocked_range<size_t>& r) {
         size_t beg = r.begin();
         size_t end = r.end();
-        std::transform(&h._hist[beg], &h._hist[end], &_hist[beg], &_hist[beg], std::plus<int>());
+        std::transform(&h._hist[beg], &h._hist[end], &_hist[beg], &_hist[beg], std::plus<size_t>());
     };
     tbb::parallel_for(range, loop);
 }
 
-void Histogram::setup_bins(const int start, const int end) {
-    int i, stride, buckets;
-    int** bins_bin;
+void Histogram::setup_bins(const size_t start, const size_t end) {
+    size_t i, stride, buckets;
+    size_t** bins_bin;
 
-    auto calc_lambda = [&](int start, int lstride) {
-        int* lbins = new int[_hist.size()];
-        int end = std::min((size_t)(start + lstride), _data.size());
-        memset(lbins, 0, _hist.size() * sizeof(int));
-        for (auto i = start; i < end; i++) {
+    auto calc_lambda = [&](size_t start, size_t lstride) {
+        size_t* lbins = new size_t[_hist.size()];
+        size_t end = std::min((size_t)(start + lstride), _data.size());
+        memset(lbins, 0, _hist.size() * sizeof(size_t));
+        for (size_t i = start; i < end; i++) {
             auto v = _data[i];
             if (std::isfinite(v)) {
-                int binN = std::max(std::min((int)((v - _min_val) / _bin_width), (int)_hist.size() - 1), 0);
+                size_t binN = std::max<size_t>(std::min<size_t>((size_t)((v - _min_val) / _bin_width), (size_t)_hist.size() - 1), 0);
```

Comment:
this doesn't match our naming convention. Please use something like `bin_number`

Reply:
(I realise you weren't responsible for the actual name :smile: )

Reply:
Fixed